Unit tests fail due to surefire plugin issue with Debian/Ubuntu Java8

See https://issues.apache.org/jira/browse/SUREFIRE-1588

The workaround is to update `maven-surefire-plugin` to version 3.0.0-M1.

```bash
$ mvn clean install
...
[INFO] --- maven-surefire-plugin:2.19.1:test (default-test) @ cannoli-core-spark2_2.11 ---

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Error: Could not find or load main class org.apache.maven.surefire.booter.ForkedBooter

Results :

Tests run: 0, Failures: 0, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Cannoli_2.11 ....................................... SUCCESS [  1.910 s]
[INFO] Cannoli_2.11: Core ................................. FAILURE [ 13.357 s]
[INFO] Cannoli_2.11: CLI .................................. SKIPPED
[INFO] Cannoli_2.11: Assembly ............................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 15.423 s
[INFO] Finished at: 2018-11-27T16:48:14-06:00
[INFO] Final Memory: 69M/938M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project cannoli-core-spark2_2.11: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test failed: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /working/cannoli/core && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -jar /working/cannoli/core/target/surefire/surefirebooter1395916330619248636.jar /working/cannoli/core/target/surefire/surefire7127060687775943490tmp /working/cannoli/core/target/surefire/surefire_02499811872893474696tmp
```

--- a/pom.xml
+++ b/pom.xml
@@ -158,7 +158,7 @@
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-surefire-plugin</artifactId>
-          <version>2.19.1</version>
+          <version>3.0.0-M1</version>
         </plugin>
         <plugin>
           <groupId>org.codehaus.mojo</groupId>

